spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java

```diff
@@ -90,6 +97,14 @@ public AbstractMqttMessageHandler(@Nullable String url, String clientId) {
 		Assert.hasText(clientId, "'clientId' cannot be null or empty");
 		this.url = url;
 		this.clientId = clientId;
+		this.clientManager = null;
+	}
+
+	public AbstractMqttMessageHandler(ClientManager<T, C> clientManager) {
+		Assert.notNull(clientManager, "'clientManager' cannot be null or empty");
+		this.clientManager = clientManager;
+		this.url = null;
+		this.clientId = null;
```

Comment:
I guess we can set both of these props from the provided `ClientManager`.
Or at least the `clientId` for some possible diagnostics purpose.

Reply:
`ClientManager` does not have any possibility to extract `clientId`/`url` as it is very generic on this abstract class level. As well as `getConnectionInfo` method - it is also parametrized.
We could add explicit `getClientId`/`getUrl` methods for `ClientManger` but to extend the interface for this purpose feels a bit too much for me.. Do you see any good ways to do it?

Reply:
Yeah... OK. I see that we can reach it via `IMqttAsyncClient.getClientId()`.
I wonder if we should delegate respectively from our `AbstractMqttMessageHandler.getClientId()`...
However let's revise this later on! We step too much aside with the current issue to solve 😄